### ISSUE TITLE
connectBlue/connectRed stay resident, connect{Red/Green/Blue}(0) just makes setColor fail to work

### ISSUE THREAD

**Issue by JessicaMulein** [sparkfun/SparkFun_Qwiic_Twist_Arduino_Library#2]
On a previous iteration I ran the example code that did connectRed(-10) and connectBlue(10).
Am trying to set a 0-100 gradient.

```
#include <Wire.h>
#include "SparkFun_Qwiic_Twist_Arduino_Library.h" //Click here to get the library: http://librarymanager/All#SparkFun_Twist
#include "Qwiic_LED_Stick.h" // Click here to get the library: http://librarymanager/All#SparkFun_Qwiic_LED_Stick

LED LEDStick;
TWIST twist;

const byte twistInterruptPin = 4; //Can be any GPIO

void setup() {
  Serial.begin(115200);

  pinMode(twistInterruptPin, INPUT_PULLUP);

  if(twist.begin() == false)
  {
    Serial.println("Twist does not appear to be connected. Please check wiring. Freezing...");
    while(1);
  }

  int currentVersion = twist.getVersion();

  if (currentVersion < 0x0201) //v1.2 in two byte form
  {
    Serial.print("The current firmware version is: ");
    Serial.print(currentVersion & 0xFF);
    Serial.print(".");
    Serial.println(currentVersion >> 8);

    Serial.println("This feature is not supported. Please consider upgrading the firmware on your Qwiic Twist. Freezing.");
    while (1); //Freeze
  }

  twist.setLimit(100); 
  twist.connectRed(0);
  twist.connectGreen(0);
  twist.connectBlue(0);
  twist.setColor(0, 0, 0); //Set Red and Blue LED brightnesses to half of max.

    //Start up communication with the LED Stick
  if (LEDStick.begin() == false){
    Serial.println("Qwiic LED Stick failed to begin. Please check wiring and try again!");
    while(1);
  }
}

void loop() {
  if (digitalRead(twistInterruptPin) == LOW)
  {
    Serial.print("Interrupt: ");

    int twistCount = twist.getCount();
  
    Serial.print("Count: ");
    Serial.print(twistCount);

    Serial.println();

    if(twist.isMoved())
    {
      Serial.print("Count:");
      Serial.print(twistCount);
    }

    if(twist.isPressed())
    {
      Serial.print("Pressed");
    }

    if(twist.isClicked())
    {
      Serial.print("Clicked!");
    }

    Serial.println();

    twist.clearInterrupts(); //Clear any interrupt bits
    twist.setColor(0, 0, (twistCount/100)*255); // DOES NOTHNG!
  }

}
```

**Comment by nseidle** [comment]
tldr;
12/255 = 0
12.0/255 = 0.047059

You're using ints to do floats. I recommend you use some Serial.prints to help you debug the color you're trying to pass. For example:

    int newColor = (twistCount / 100) * 255;

    Serial.print("newColor: ");
    Serial.print(newColor);

    twist.setColor(0, 0, newColor);

If you run the above code you'll see now that the output is wrong because the compiler is dividing twistCount (a uint8_t) by 100 (an int arguably). Said differently 12/255 = 0 not 0.047059. To fix you can cast one of the variables as a float or you can just force a float into the math like this

    int newColor = (twistCount / 100.0) * 255;

**Comment by JessicaMulein** [comment]
Thanks for the great catch @nseidle !

**Comment by nseidle** [comment]
Np! Good luck with the project.
